ojdbc-provider-jackson-oson/src/main/java/oracle/jdbc/provider/oson/OsonGenerator.java

```diff
@@ -0,0 +1,509 @@
+    int status = _writeContext.writeValue();
+    if(status != _writeContext.STATUS_OK_AFTER_COMMA) {
+      _reportError("error: expecting end array");
+    }
+    if (!_writeContext.inArray()) {
+      _reportError("Current context not an ARRAY but " + _writeContext.getTypeDesc());
+    }
+    gen.writeEnd();
+    _writeContext = _writeContext.getParent();
+  }
+
+  /**
+   * Writes the start of a JSON object.
+   *
+   * @throws IOException If an I/O error occurs.
+   */
+  @Override
+  public void writeStartObject() throws IOException {
+    _verifyValueWrite("write start object");
+    _writeContext = _writeContext.createChildObjectContext();
+    gen.writeStartObject();
+  }
+
+  /**
+   * Writes the end of a JSON object.
+   *
+   * @throws IOException If an I/O error occurs.
+   */
+  @Override
+  public void writeEndObject() throws IOException {
+    int status = _writeContext.writeValue();
+    if (status != JsonWriteContext.STATUS_EXPECT_NAME) {
+      _reportError("error: expecting end object");
+    }
+    if (!_writeContext.inObject()) {
+      _reportError("Current context not an OBJECT but " + _writeContext.getTypeDesc());
+    }
+    gen.writeEnd();
+    _writeContext = _writeContext.getParent();
+  }
+
+  /**
+   * Writes a field name within a JSON object.
+   *
+   * @param name The field name to write.
+   * @throws IOException If an I/O error occurs.
+   */
+  @Override
+  public void writeFieldName(String name) throws IOException {
+    int status = _writeContext.writeFieldName(name);
+    if (status == JsonWriteContext.STATUS_EXPECT_VALUE) {
+      _reportError("Can not write a field name, expecting a value");
+    }
+    gen.writeKey(name);
+  }
+
+  /**
+   * Writes a string value.
+   *
+   * @param text The string to write.
+   * @throws IOException If an I/O error occurs.
+   */
+  @Override
+  public void writeString(String text) throws IOException {
+    _verifyValueWrite("write string");
+    gen.write(text);
+  }
+
+  /**
+   * Writes a string value from a character array.
+   *
+   * @param buffer The character array containing the string.
+   * @param offset The starting offset in the array.
+   * @param len The number of characters to write.
+   * @throws IOException If an I/O error occurs.
+   */
+  @Override
+  public void writeString(char[] buffer, int offset, int len) throws IOException {
+    _verifyValueWrite("write string");
+    gen.write(new String(buffer, offset, len));
+  }
+
+  /**
+   * Writes a string from a byte array using UTF-8 encoding.
+   *
+   * @param buffer The byte array containing the string.
+   * @param offset The starting offset in the array.
+   * @param len The number of bytes to write.
+   * @throws IOException If an I/O error occurs.
+   */
+  @Override
+  public void writeRawUTF8String(byte[] buffer, int offset, int len) throws IOException {
+    _verifyValueWrite("writeRawUTF8String");
+    gen.write(new String(buffer, offset, len));
```

Comment:
How can we be sure that UTF-8 encoding is being used? According to the javadoc, this constructor constructs a new String by decoding the specified subarray of bytes using the platform's default charset.